src/crimson/admin/osd_admin.cc

```diff
@@ -231,20 +231,35 @@ class DumpMetricsHook : public AdminSocketHook {
 
   static void dump_metric_value(Formatter* f,
                                 string_view full_name,
-                                const registered_metric& metric)
+                                const registered_metric& metric,
+                                const seastar::metrics::impl::labels_type& labels)
   {
+    std::ostringstream oss_name;
+    oss_name << full_name << "(";
```

Comment:
if we really want to leverage labels as part of the naming scheme, i'd suggest use a more structured approach.

1. to support syntax like `ceph dump-metrics metric_name label_name_a=label_value_a label_name_b=label_value_b...`, so we can query and print a certain metric with specified label kv pairs.
2. dump each metric like an object, like `{"value": 123, "label_name_a": "label_value_a", ...}`. so one is able to pipe the output to `jq` for further processing

Reply:
I'll modify the dump format to fit the structure as suggested, for example.
`cache_trans_committed{"shard": 0, "src": "CLEANER", "value": 1}", ...`

Reply:
see https://github.com/ceph/ceph/pull/42202/commits/6b38bf1f1be71238d45185433c066b4476c1f500